controllers/builder/openshiftbuilder.go

```diff
@@ -0,0 +1,273 @@
+								ConfigMap: corev1.LocalObjectReference{},
+							},
+						},
+					*/
+				},
+				Strategy: buildv1.BuildStrategy{
+					Type: buildv1.DockerBuildStrategyType,
+					DockerStrategy: &buildv1.DockerBuildStrategy{
+						ImageOptimizationPolicy: &optimizationPol,
+					},
+				},
+				Output: buildv1.BuildOutput{
+					To: &v1.ObjectReference{
+						Namespace: build.Namespace,
+						Name:      build.Status.ImageTag,
+						Kind:      imageStreamTagKind,
+					},
+				},
+				Resources: build.Spec.Resources,
+			},
+		},
+	}
+}
+
+func (o *openshiftBuilderManager) Reconcile(build *operatorapi.KogitoServerlessBuild) (err error) {
+	var openshiftBuild *buildv1.Build
+
+	if build.Status.BuildPhase == operatorapi.BuildPhaseNone ||
+		build.Status.BuildPhase == operatorapi.BuildPhaseInitialization {
+
+		// guard to avoid spamming multiple builds
+		if openshiftBuild, err = o.fetchOpenShiftBuildRef(build); err != nil {
+			return err
+		}
+		if openshiftBuild.Status.Phase == buildv1.BuildPhaseNew ||
+			openshiftBuild.Status.Phase == buildv1.BuildPhaseRunning ||
+			openshiftBuild.Status.Phase == buildv1.BuildPhasePending {
+			build.Status.BuildPhase = operatorapi.BuildPhaseRunning
+			return nil
+		}
+
+		// we push another one
+		workflow, err := o.fetchWorkflowForBuild(build)
+		if err != nil {
+			return err
+		}
+		// kick a new build
+		openshiftBuild, err = o.pushNewOpenShiftBuildForWorkflow(build, workflow)
+		if err != nil {
+			return err
+		}
+		build.Status.BuildPhase = operatorapi.BuildPhaseScheduling
+		return build.Status.SetInnerBuild(kubeutil.ToTypedLocalReference(openshiftBuild))
+	}
+
+	if openshiftBuild, err = o.fetchOpenShiftBuildRef(build); err != nil {
+		return err
+	}
+	if openshiftBuild == nil {
+		build.Status.BuildPhase = operatorapi.BuildPhaseInitialization
+		return nil
+	}
+
+	// Checks the phase
+	build.Status.BuildPhase = openshiftBuildPhaseMatrix[openshiftBuild.Status.Phase]
+	if openshiftBuild.Status.Phase == buildv1.BuildPhaseError {
+		build.Status.Error = openshiftBuild.Status.Message
+	}
+
+	return build.Status.SetInnerBuild(kubeutil.ToTypedLocalReference(openshiftBuild))
+}
+
+func (o *openshiftBuilderManager) fetchOpenShiftBuildRef(build *operatorapi.KogitoServerlessBuild) (*buildv1.Build, error) {
+	openshiftBuild := &buildv1.Build{}
+	refOpenShiftBuild := &v1.TypedLocalObjectReference{}
+	if err := build.Status.GetInnerBuild(refOpenShiftBuild); err != nil {
+		return nil, err
+	}
+	if err := o.client.Get(o.ctx, types.NamespacedName{Name: refOpenShiftBuild.Name, Namespace: build.Namespace}, openshiftBuild); err != nil {
+		if errors.IsNotFound(err) {
+			return openshiftBuild, nil
+		}
+		return nil, err
+	}
+	return openshiftBuild, nil
+}
+
+// TODO: this should be from fileS, in this case we can TAR everything in a temp directory within the operator pod fs and push
+// TODO: for now, we mount the CMs from the devmode into the build and push only the bytes for the workflow definition from memory
```

Comment:
jira?

Reply:
Not opened yet. UI will do it, should be solved this sprint.